app/src/main/java/com/github/se/stepquest/StepGoalSetBox.kt

```diff
@@ -0,0 +1,137 @@
+                    placeholder = { Text("5000") },
+                    keyboardOptions =
+                        KeyboardOptions.Default.copy(
+                            keyboardType = KeyboardType.NumberPassword, imeAction = ImeAction.Done),
+                    keyboardActions =
+                        KeyboardActions(
+                            onDone = {
+                              val dailyStep =
+                                  newDailyStepGoal
+                                      .filter { it.isDigit() }
+                                      .take(5)
+                                      .let {
+                                        if (it.isBlank()) {
+                                          5000 // Default value if blank
+                                        } else {
+                                          val parsedInput = it.toIntOrNull() ?: 0
+                                          val roundedValue = (parsedInput + 249) / 250 * 250
+                                          if (roundedValue < 1000) {
+                                            1000
+                                          } else {
+                                            roundedValue
+                                          }
+                                        }
+                                      }
+                              val weeklyStep =
+                                  newWeeklyStepGoal.takeIf { it.isNotBlank() }?.toInt()
+                                      ?: (dailyStep * 7)
+                              onConfirm(dailyStep, weeklyStep)
+                              onDismiss()
+                            }))
+                Spacer(modifier = Modifier.height(16.dp))
+                Button(
+                    onClick = {
+                      val dailyStep =
+                          newDailyStepGoal
+                              .filter { it.isDigit() }
+                              .take(5)
+                              .let {
+                                if (it.isBlank()) {
+                                  5000 // Default value if blank
+                                } else {
+                                  val parsedInput = it.toIntOrNull() ?: 0
+                                  val roundedValue = (parsedInput + 249) / 250 * 250
+                                  if (roundedValue < 1000) {
+                                    1000
+                                  } else {
+                                    roundedValue
+                                  }
+                                }
+                              }
+                      val weeklyStep =
+                          newWeeklyStepGoal.takeIf { it.isNotBlank() }?.toInt() ?: (dailyStep * 7)
+                      onConfirm(dailyStep, weeklyStep)
+                      onDismiss()
+                    },
+                    colors = ButtonDefaults.buttonColors(blueThemeColor),
```

Comment:
Lines 81-102 and 107-127 are exactly the same, maybe make it more modular/find a way to write it only once for better readability? Otherwise looks good!